test/integration/framework/test_server.go

```diff
@@ -228,3 +233,10 @@ func StartTestServer(t testing.TB, setup TestServerSetup) (client.Interface, *re
 
 	return kubeAPIServerClient, kubeAPIServerClientConfig, tearDownFn
 }
+
+type checkNothing struct{}
```

Comment:
The alternative is (untested):
```
func init() {
   _ := healtz.LogHealthz.Check(nil)
}
```

That starts the goroutine during init and then `goleak.IgnoreCurrent` knows about it.

I'm undecided whether that is a better solution than replacing the instance.

Reply:
can't we plumb a context on the handler to stop it on apiserver shutdown?

Reply:
it seems doesn't work

```
	found unexpected goroutines:
	[Goroutine 2263 in state select, with k8s.io/kubernetes/vendor/k8s.io/apimachinery/pkg/util/wait.BackoffUntil on top of the stack:
	goroutine 2263 [select]:
	k8s.io/kubernetes/vendor/k8s.io/apimachinery/pkg/util/wait.BackoffUntil(0x59f07c8?, {0x59be860, 0xc0092efbf0}, 0x1, 0xc0000925a0)
		/home/prow/go/src/k8s.io/kubernetes/_output/local/go/src/k8s.io/kubernetes/vendor/k8s.io/apimachinery/pkg/util/wait/wait.go:169 +0x135
	k8s.io/kubernetes/vendor/k8s.io/apimachinery/pkg/util/wait.JitterUntil(0x59c0fa0?, 0xdf8475800, 0x0, 0x25?, 0xc00317ef00?)
		/home/prow/go/src/k8s.io/kubernetes/_output/local/go/src/k8s.io/kubernetes/vendor/k8s.io/apimachinery/pkg/util/wait/wait.go:135 +0x89
	k8s.io/kubernetes/vendor/k8s.io/apimachinery/pkg/util/wait.Until(...)
		/home/prow/go/src/k8s.io/kubernetes/_output/local/go/src/k8s.io/kubernetes/vendor/k8s.io/apimachinery/pkg/util/wait/wait.go:92
	k8s.io/kubernetes/vendor/k8s.io/apimachinery/pkg/util/wait.Forever(0x0?, 0x0?)
		/home/prow/go/src/k8s.io/kubernetes/_output/local/go/src/k8s.io/kubernetes/vendor/k8s.io/apimachinery/pkg/util/wait/wait.go:83 +0x28
	created by k8s.io/kubernetes/vendor/k8s.io/apiserver/pkg/server/healthz.(*log).Check.func1
		/home/prow/go/src/k8s.io/kubernetes/_output/local/go/src/k8s.io/kubernetes/vendor/k8s.io/apiserver/pkg/server/healthz/healthz.go:74 +0xe5
	]
```

Reply:
> can't we plumb a context on the handler to stop it on apiserver shutdown?

No, healthz.go initializes the variable during init. Goroutine startup then gets delayed until used for the first time.

> it seems doesn't work

I'll check that tomorrow. So far, I was testing only with `test/integration/scheduler_perf`. `make test-integration` fails for me for various reasons.

Reply:
We have multiple copies of `StartTestServer`. I modified the one in `test/integration/framework/test_server.go`, but not the one in `cmd/kube-apiserver/app/testing/`... let's do this all in etcd.go where the leak checking is done.